pysyncobj/utility.py

```diff
@@ -0,0 +1,79 @@
+import os
+import time
+
+from .encryptor import getEncryptor
+from .node import Node, TCPNode
+from .poller import createPoller
+from .tcp_connection import TcpConnection
+
+
+class TcpUtility(object):
+
+    def __init__(self, password=None, timeout=900.0):
+        self.__timeout = timeout
+        self.__poller = createPoller('auto')
+        self.__connection = TcpConnection(self.__poller,
+                                          onDisconnected=self.__onDisconnected,
+                                          onMessageReceived=self.__onMessageReceived,
+                                          onConnected=self.__onConnected,
+                                          timeout=timeout)
+        if password is not None:
+            self.__connection.encryptor = getEncryptor(password)
+
+        self.__result = None
+        self.__error = None
+
+    def executeCommand(self, node, message):
+        self.__result = None
+        self.__error = None
+
+        if not isinstance(node, Node):
+            try:
+                node = TCPNode(node)
+            except Exception:
+                self.__error = 'invalid address to connect'
+                return
+
+        self.__isConnected = self.__connection.connect(node.host, node.port)
+        if not self.__isConnected:
+            self.__error = "can't connected"
+            return
+
+        deadline = time.time() + self.__timeout
+
+        self.__data = message
+        while self.__isConnected:
+            self.__poller.poll(0.5)
+            if time.time() > deadline:
+                self.__connection.disconnect()
+
+        return self.getResult() or self.getError()
```

Comment:
May be at least return tuple / dict? How can we distinguish errors from non-errors? (this is optional, if you don't think it's a good idea or don't have time - I'll merge as is)

Reply:
Well, yeah, this is the issue.
That's why in the original code I was explicitly checking if the result is None and returning the error.
Unfortunately, I don't have a better interface for that, except maybe raising an exception.

Reply:
What about ```return (self.getResult(), self.getError()) ``` ?

Reply:
I think it would be more appropriate to raise an exception.
After all, this is what exceptions are useful for :)

Reply:
Ok